csrc/predicate_compute.cpp

```diff
@@ -375,6 +375,166 @@ std::size_t UnswitchPredicateKeyHash::operator()(
   return h;
 };
 
+namespace {
+
+// Select first warp of threads along TIDx axis and then use ptx::elect_sync
+// TODO If TIDx is known at compile-time, generate custom mask.
+Val* createElectSyncPredicate() {
+  Val* warp_size = IrBuilder::create<Val>(32L, PrimDataType::UInt64);
+  Val* full_mask_val = IrBuilder::create<Val>(0xFFFFFFFF, PrimDataType::UInt32);
+  Val* elect_sync_val = IrBuilder::create<Val>(PrimDataType::Bool);
+  IrBuilder::create<UnaryOp>(
+      UnaryOpType::ElectSync, elect_sync_val, full_mask_val);
+  return SimplifyingIrBuilder::logicalAndExpr(
+      elect_sync_val,
+      IrBuilder::ltExpr(
+          NamedScalar::getParallelIndex(ParallelType::TIDx), warp_size));
+}
+
+Val* createElectSyncPredicate(kir::Predicate* pred) {
+  NVF_ERROR(pred != nullptr);
+  NVF_ERROR(pred->expr() != nullptr);
+
+  TensorView* out_tv = ir_utils::getTvOutput(pred->expr());
+  NVF_ERROR(out_tv != nullptr, "Missing TensorView output");
+
+  bool is_tv_tidx_parallelized = std::any_of(
+      out_tv->domain()->loop().begin(),
+      out_tv->domain()->loop().end(),
+      [](IterDomain* id) {
+        return id->getParallelType() == ParallelType::TIDx;
+      });
+
+  // short-circuit: out_tv uses ParallelType::TIDx
+  if (is_tv_tidx_parallelized) {
+    return pred->fusion()->trueVal();
+  }
```

Comment:
Am I understanding correct that we are generating something like:
```CUDA
if (elect-sync) {
  if (regular predicate) {
    expr;
  }
}
```
where the `elect-sync` predicate will be `true` for the case when it does have TIDx parallelization?

Reply:
Yes, the `elect-sync` predicate will be `true` when the expression is `TIDx` parallelization.
e.g., You're using the the `TIDx` dimension to launch multiple expressions. There shouldn't be a need to select an arbitrary thread.

I suppose you could use `elect-sync` on another dimension, but this was enough for now.